internal/serialization/class_definition_context.go

```diff
@@ -28,32 +29,39 @@ import (
 
 type ClassDefinitionContext struct {
 	classDefs map[string]*serialization.ClassDefinition
+	mu        *sync.RWMutex
 	factoryID int32
 }
 
 func NewClassDefinitionContext(factoryID int32) *ClassDefinitionContext {
 	return &ClassDefinitionContext{
 		factoryID: factoryID,
 		classDefs: make(map[string]*serialization.ClassDefinition),
+		mu:        &sync.RWMutex{},
 	}
 }
 
 func (c *ClassDefinitionContext) LookUp(classID int32, version int32) *serialization.ClassDefinition {
-	return c.classDefs[encodeVersionedClassID(classID, version)]
+	c.mu.RLock()
+	cd := c.classDefs[encodeVersionedClassID(classID, version)]
+	c.mu.RUnlock()
```

Comment:
Is there any difference between calling `c.mu.RUnlock()` directly or in `defer` statement in this function?

Reply:
Same comment applies PortableContext.LookUp too.

Reply:
Thanks.